external/kafka-0-10-sql/src/test/scala/org/apache/spark/sql/kafka010/KafkaMicroBatchSourceSuite.scala

```diff
@@ -1511,6 +1515,54 @@ abstract class KafkaSourceSuiteBase extends KafkaSourceTest {
       CheckAnswer(2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17)
     )
   }
+
+  protected def testNullableKeyValue(trigger: Trigger): Unit = {
+    val table = "kafka_null_key_value_source_test"
+    withTable(table) {
+      val topic = newTopic()
+      testUtils.createTopic(topic)
+      testUtils.withProducer { producer =>
```

Comment:
can't we use `withTranscationalProducer`?

Reply:
We can, but is there any strong reason to test with kafka transcational producer?

Reply:
I worry testing with a kafka transcational producer may imply some special reason. Maybe i want more.

Reply:
because we add a new method `withProducer`, and we need to justify it by explaining why `withTranscationalProducer` doesn't work here.

Reply:
@cloud-fan Dont strongly object `withTranscationalProducer`, but not very clear why we should test with `withTranscationalProducer`. @gaborgsomogyi has explained, and make sense.